tests/test-http-client-transport/src/test/java/org/eclipse/jetty/http/client/HttpClientContinueTest.java

```diff
@@ -507,35 +512,40 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
                 public void onComplete(Result result)
                 {
                     assertArrayEquals(data, getContent());
-                    latch.countDown();
+                    requestLatch.countDown();
                 }
             });
 
-        Thread.sleep(1000);
+        // Wait for the handler thread to be blocked in IO.
+        await().atMost(5, TimeUnit.SECONDS).until(() ->
+        {
+            Thread thread = handlerThread.get();
+            return thread != null && thread.getState() == Thread.State.WAITING;
+        });
 
         content.offer(ByteBuffer.wrap(chunk1));
 
-        Thread.sleep(1000);
+        assertTrue(demandLatch.await(5, TimeUnit.SECONDS));
```

Comment:
Before, the 1 second sleep kinda made sure the first chunk arrived to the server.
Now, the 2 chunks may arrive to the server in one chunk, as the wait has been practically zeroed, so the test is now different from before.

Reply:
* await on client until read() blocks.
* read exactly chunk1.length bytes
* now the next read on server will block, count down a latch to tell main thread to await again
* block on read() on server
* await until read() blocks
* offer second chunk.
* read second chunk on server
* wait for successful response on client